action.yml

```diff
@@ -9,6 +9,11 @@ inputs:
       Space-separated list of tasks to run
     required: false
     default: ""
+  additional-build-options:
+    description: >-
+      Additional build options for MATLAB build tool
```

Comment:
Those sound good to me! 

I was initially staying away from `build-option` due to potential confusion with the overall GitHub Actions pipeline, but I think the fact that in context, `build-options` will be scoped to within the `with` block under the `run-build` action will make it clear what type of build it's referring to.

Reply:
I am planning to go ahead with design which includes a combination of some dedicated parameter names and a common build options parameter. But as it stands right now, the team isn't favoring addition of either `skip` or `continueOnFailure` to have their on dedicated parameters. Even though we won't be adding any parameters for build options right now but keeping in mind that it might change in future I chose `additional-build-options` for now. If something like `extra-build-options` sounds better, we can go ahead with that as well.

I was initially considering `build-options` as the common parameter name but if in future we introduce a new dedicated parameter for some build option, say `verbosity`, we wouldn't want to update our parameter name from `build-options` to `additional-build-options` at that point in time. When we'd be adding a `verbosity` parameter, if keeping the common parameter name `build-options` as it is would be a better choice than current implementation, I can wrap my head around that.

Reply:
I agree with Kapil that removing the `additional` or some variant like `extra` doesn't work in the design he is moving towards (his design which included some but not all options "promoted" to the top level). 

If there were no promoted build options then I agree those words are extra (haha) and unnecessary, but I think something is needed in the hybrid case, which ultimately Kapil has chosen after receiving the team's feedback on it.

Having `additional` in the name without any promoted options also feels odd.

My take is that we promote an option (continueOnFailure?) so that we have at least one promoted option and then the relief valve for others not yet (or ever) promoted. Then I would call it `additional-options` or `extra-options`. The `build` is not needed since it is known from the context. 

If we dont want to promote anything yet, I do think we would need some qualifier that works in both the case where an option is promoted and when there are not yet any promoted options. I haven't been able to think of what such a qualifier would be beyond the two that Kapil is considering.

Reply:
@mw-kapilg That makes a lot of sense in the naming discussion. I'm supportive of Andy's suggestion of `additional-options` in that case, although I think we'd have to be fairly clear in the documentation that this category was a passthrough for MBT options.

I also don't think I particularly mind the pass through being called `build-options` anyway, since it's still a fairly descriptive name for what the field will be doing even after we break out some of the other build options to the top level. What are people's thoughts on 

```
with:
    verbosity: 4
    continue-on-failure: true
    additional-options: -skip test
```

vs

```
with:
    verbosity: 4
    continue-on-failure: true
    build-options: -skip test
```

I'm not sure that the latter is any more confusing than the former, especially since even after we add the break out options they won't exclude the user from specifying those settings via `additional-options` / `build-options`.

I'm supportive of whatever choice you end up making @mw-kapilg , I think all of the ideas mentioned in this conversation are pretty solid.

Reply:
Adding @mcafaro to the thread.

Reply:
I think if we choose to remove build and keep it `additional-options`, there might be a portion of users who'd get confused when they see it along with `startup-options`. Something like this:
```
uses: matlab-actions/run-build
with:
    startup-options: -nojvm
    additional-options: -skip test
```

@acampbel : Not everybody is here

Reply:
I thought the input equally represents all the options of the buildtool command. Separating the build options (ContinueOnFailure vs. the rest) is somewhat confusing to me.

Reply:
@mw-hrastega In case of design 3 [here](https://confluence.mathworks.com/pages/viewpage.action?spaceKey=CI&title=Build+options+for+CI+integrations#BuildoptionsforCIintegrations-Design3:CombineDesigns1&2(preferred)), the input is supposed to take in all other options which don't have a dedicated parameter. Note that if the user decided to use a flag using the common input even though it has its own parameter, the action would still run fine.
Having a common input like `additional-build-options` (in addition to some promoted parameters) would also make the user wonder what else is available as build option. And in turn they'd be able to use latest build options, for e.g. -parallel as it gets released with 24a, without us needing to add a dedicated parameter to our plugin.

Reply:
Thanks for clarifying @mw-kapilg. My question is mostly around the criteria we will use to promote certain options. How will we decide which options deserve their own inputs? As for the buildtool command, all the options are treated equally, so not sure how we will differentiate in the CI context.

Reply:
For now, we were considering **usage** as a filter to decide whether it should be promoted or not. Open to suggestions on other filters!
Thinking of it now, I'm not sure if the team voted on the poll based on usage. Wondering what else could have been a factor they voted on 🤔

Reply:
Good point! Right now we maintain a flat list of options on the [buildtool page](https://www.mathworks.com/help/matlab/ref/buildtool.html#mw_50c0f35e-93df-4579-963d-f59f2fba1dba). It will be nice to get some insights into the usage and importance of various build options.

Reply:
All seem like good points. I personally prefer `build-options` to `additional-options` because of `startup-options`. 

A bit of a tossup on `build-options` vs `additional-build-options` to me. I think the "additional" part can be handled by the help text but not opposed to adding it to the input name for additional clarity.

I can also see a future where we never promote any dash options because they are easy enough to just put in `build-options`. This is not true with `run-tests` where the options are more difficult to achieve with the underlying API. 

A potential threshold for adding a dedicated checkbox/text field might be:
- can the action provide additional value by knowing that this option is set and/or what it's value is?
- can this option not be configured via a simple dash arg?

Reply:
Nice. I also prefer "build-options" because it keeps up safe in case we never promote any dash options.

Reply:
I like the idea of giving clarity via the help text. Seems to be a decent middle ground for both cases whether we have any promoted options or not. I'll go ahead and update the the parameter name to `build-options`, let me know if someone has a flag with that. Thanks everybody for being part of the discussion!